client.go

```diff
@@ -394,6 +387,12 @@ func (c *Client) runOnce() error {
 
 	go c.runPublisher(outputCh, &wg)
 
+	// Notify everyone that the client has started. Runs sequentially so there
+	// isn't any race conditions when working with the connections or channels.
+	for _, onStarted := range c.onStarteds {
+		onStarted(inputConn, outputConn, inputCh, outputCh)
+	}
```

Comment:
Which makes me wonder if we should've called this `OnConnected`... Are you planning a new major? If so we can rename it, otherwise probably not worth it.

Reply:
Yea, lets do that!